### DISCUSSION TITLE
网络图片的识别，多进程，会导致tmp.jpg被反复覆盖

### DISCUSSION THREAD

**Discussion by kylin987** [PaddlePaddle/PaddleOCR#13038] [Q&A]
建议增加参数，或者tmp.jpg不要写死

**Comment by kylin987** [comment]
多进程下，会出现，A进程的图片被B进程的覆盖，从而导致A进程识别了B的图片

**Comment by tink2123** [comment]
请问是测试哪个任务遇到的问题呢？

**Comment by kylin987** [comment]
普通的ocr识别
ocr = PaddleOCR(use_angle_cls=True, lang="ch")  # need to run only once to download and load model into memory
img_path = '网络图片'
result = ocr.ocr(img_path, cls=True)

**Comment by tink2123** [comment]
收到～ 感谢您的反馈，可以先自行修改下whl包的代码，网络图片保存为本地图片时用进程id区分。

**Comment by kylin987** [comment]
不客气，最好增加参数，允许配置tmp的文件名，也为了后续删除图片做准备。有的业务识别后，图片也会删除的

**Comment by tink2123** [comment]
> 不客气，最好增加参数，允许配置tmp的文件名，也为了后续删除图片做准备。有的业务识别后，图片也会删除的

好的～ 记录下您的反馈了

**Comment by kylin987** [comment]
您说的进程id我仔细想了下，比传参数可能更好，反正就是像之前一样反复覆盖，用进程id保证了进程间不会互相干扰

**Comment by liyuweihuo** [comment]
请问你使用超过2个的多进程并发，单个预测时间有没有翻倍，开5个并发，单个图片文字识别直接翻倍，cpu使用率上不去。cpu环境
